test/IRGen/weaklinked_import.swift

```diff
@@ -0,0 +1,159 @@
+// RUN: %empty-directory(%t)
```

Comment:
Could you also add a test for adding `@_weakLinked import Foo` where `Foo` is a transitive dependency? APIs from `Foo` are not directly used by the source file, but a direct dependency `Bar` of the source file has some APIs that trigger inlining `Foo`'s APIs. Those APIs should be weakly linked.

Reply:
Good idea, I'll add that test.

Reply:
This was a good catch. It turns out that the logic in this PR does not properly change the linkage of symbol references inlined from other modules. I think it's going to require some deeper investigation so I'd like to tackle it in another PR.

Reply:
Sounds good!

Reply:
I added a short test case that documents that this currently doesn't work.